app/dashboard/src/components/AriaComponents/Inputs/Password/Password.tsx

```diff
@@ -0,0 +1,57 @@
+/** @file A component wrapping {@link Input} with the ability to show and hide password. */
+import { useState } from 'react'
+import type { Path } from 'react-hook-form'
+
+import EyeIcon from '#/assets/eye.svg'
+import EyeCrossedIcon from '#/assets/eye_crossed.svg'
+import {
+  Button,
+  Input,
+  type FieldValues,
+  type InputProps,
+  type TSchema,
+} from '#/components/AriaComponents'
+
+// ================
+// === Password ===
+// ================
+
+/** Props for a {@link Password}. */
+export interface PasswordProps<
+  Schema extends TSchema,
+  TFieldValues extends FieldValues<Schema>,
+  TFieldName extends Path<TFieldValues>,
+  TTransformedValues extends FieldValues<Schema> | undefined = undefined,
+> extends Omit<
+    InputProps<Schema, TFieldValues, TFieldName, TTransformedValues>,
+    'addonMiddle' | 'type'
+  > {}
+
+/** A component wrapping {@link Input} with the ability to show and hide password. */
+export function Password<
+  Schema extends TSchema,
+  TFieldValues extends FieldValues<Schema>,
+  TFieldName extends Path<TFieldValues>,
+  TTransformedValues extends FieldValues<Schema> | undefined = undefined,
+>(props: PasswordProps<Schema, TFieldValues, TFieldName, TTransformedValues>) {
+  const [showPassword, setShowPassword] = useState(false)
+
+  return (
+    <Input
+      {...props}
+      type={showPassword ? 'text' : 'password'}
+      addonMiddle={
```

Comment:
Small question: why do we need `addonMiddle`, why neither `addonStart` nor `addonEnd` aren't sufficient here?

Reply:
a) we need the eye to be centered on the text input. maybe we always know the height so we can position using `rem` without having to use `top-1/2`?
b) i was working under the assumption that users of `<Password` may want to add their own `addonStart`/`addonEnd`, although it's true that doing this still prevents them from adding their own `addonMiddle`

Reply:
A. Looks like the naming "addonStart" and "addonEnd" is misleading 🥲 start and end here means "left" and "right", but they _might_ flip based on the locale(eg for RTL languages). Maybe there's a better naming for that?

B. we can use React.Fragment and stack multiple addons

Reply:
oh... yeah, makes sense. i'll mark as resolved since this is addressed locally but open an issue since i do think the naming is a bit confusing